source/lambda/schedulers/ec2_service.py

```diff
@@ -75,15 +75,15 @@ class Ec2Service:
     EC2_STOPPING_STATES = {EC2_STATE_SHUTTING_DOWN, EC2_STATE_STOPPING, EC2_STATE_STOPPED}
     EC2_STARTING_STATES = {EC2_STATE_PENDING, EC2_STATE_RUNNING}
 
-    maintenance_table = DynamoDBUtils.get_dynamodb_table_resource_ref(os.environ['MAINTENANCE_WINDOW_TABLE'])
-
     def __init__(self):
         self.service_name = "ec2"
         self.allow_resize = True
         self.schedules_with_hibernation = []
         self._ssm_maintenance_windows = None
         self._session = None
         self._logger = None
+        self._dynamodb = boto3.resource('dynamodb')
```

Comment:
Remember that `resource` isn't receiving updates, I'd avoid building more on top of it if at all possible

Reply:
ec2_service (and related scheduler code) is currently highly fragile and not open to easy modification. The purpose of all these changes to the testing/build process is to allow for tests to be written against these files especially so that edits exactly like you mention can be done safely